x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/cache/SparseFileTracker.java

```diff
@@ -330,12 +399,17 @@ public String toString() {
             this.end = end;
         }
 
-        @Override
-        public void onResponse(Void aVoid) {
+        public void onCompletion() {
             onGapSuccess(start, end);
         }
 
-        @Override
+        public void onProgress(long value) {
+            if (value < start || end < value) {
+                throw new IllegalArgumentException("Cannot update progress [" + value + "] for gap [" + start + '-' + end + ']');
```

Comment:
I think this shouldn't happen, so should be an assertion. Maybe just remove it since we assert it in `onGapProgress` anyway.